frontend/src/components/cronjob/Details.tsx

```diff
@@ -54,6 +71,103 @@ export default function CronJobDetails() {
 
   const ownedJobs = filterOwnedJobs(jobs);
 
+  // method to generate a unique string
+  const uniqueString = () => {
+    const timestamp = Date.now().toString(36);
+    const randomNum = Math.random().toString(36).substr(2, 5);
+    return `${timestamp}-${randomNum}`;
+  };
+
+  function SpawnJobDialog() {
```

Comment:
I'd declare this dialog outside of the CronJobDetails if possible.

Reply:
updated